#### 1. 
Find the equation of the regression line for the given points. Round any final values to the nearest hundredth, if necessary:
( 5.6, 8.8 ), ( 6.3, 12.4 ), ( 7, 14.8 ), ( 7.7, 18.2 ), ( 8.4, 20.8 )

In [2]:
import statsmodels.api as sm
import numpy as np
import pandas as pd

In [11]:
x = [5.6, 6.3, 7, 7.7, 8.4]
y = [8.8, 12.4, 14.8, 18.2, 20.8]
d = {'X': x, 'Y':y}
df = pd.DataFrame(data=d)
type(df)

pandas.core.frame.DataFrame

In [12]:
df

X     Y
0  5.6   8.8
1  6.3  12.4
2  7.0  14.8
3  7.7  18.2
4  8.4  20.8

In [22]:
x = sm.add_constant(x) # let's add an intercept (beta_0) to our model
model  = sm.OLS(y,x).fit()
predictions = model.predict(x) # make the predictions by the model

# Print out the statistics
model.summary()

/home/nbuser/anaconda3_501/lib/python3.6/site-packages/statsmodels/stats/stattools.py:72: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  "samples were given." % int(n), ValueWarning)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.996
Model:                            OLS   Adj. R-squared:                  0.995
Method:                 Least Squares   F-statistic:                     843.1
Date:                Fri, 06 Dec 2019   Prob (F-statistic):           8.97e-05
Time:                        18:22:27   Log-Likelihood:               -0.19107
No. Observations:                   5   AIC:                             4.382
Df Residuals:                       3   BIC:                             3.601
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        -14.8000      1.037    -14.278      0.001     -18.099     -11.501
x1             4.2571      0.147     29.036      0.000       3.791       4.724
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   3.296
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.714
Skew:                           0.498   Prob(JB):                        0.700
Kurtosis:                       1.440   Cond. No.                         51.5
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Interpreting the Table — With the constant term the coefficients are different. With a constant we are forcing our model to go ta y-intercept at -14.8, this would have the the slope of the x1 predictor as 4.2571.

Equation of the line can be written as: 
y = -14.8 + 4.25 x



#### 2. 
Find all local maxima, local minima, and saddle points for the function given below. Write your answer(s) in the form ( x, y, z ). Separate multiple points with a comma.
$ f(x,y)=24x - 6xy^2 -8y^3 $

 

$$ f(x,y) =24x - 6xy^2 -8y^3  \\
\textrm{we will calcualte partial $\partial$ derivative of f(x,y) wrt x and y. } \\
\frac{\partial f}{\partial x} = 24-6y^2 ......\textrm{ keeping y as constant } .....=>eq(1)\\ 
=> 24-6y^2 = 0 ...\textrm{ (f(x)=0 for ctrical point )}  \\
=>y= \sqrt{4} = \pm 2 ..\\
\frac{\partial f}{\partial y} =-12xy-24y^2 ......\textrm{ keeping x as constant }.....=>eq(2)\\ 
\textrm{ using y=$\pm$ 2 in eq 2 } \\
y = 2,  x = -2y ,  x = -4 \\
y = -2,  x = -2y ,  x = 4 \\
\textrm{ We have two critical point, lets check using 2nd derivative to find if its maxima or minima or saddle point}
(4,-2) and (-4,2)
$$

Calculating 2nd Derivative 

$$
\frac{\partial^2 f}{\partial x^2} =0 \\
\frac{\partial^2 f}{\partial y^2} =-12x-48y \\
\frac{\partial^2 f}{\partial x \partial y} = -12y \\
$$

---------------------------------------------------------------------
|  ....Derivatives \ Point.... | (4,-2)  |  (-4,2) |   |   |
|:---------------------------------------------------------------------:|---|---|---|---|
|  $\frac{\partial f}{\partial x}$                                    | 0  |   0|   |   |
|  $\frac{\partial f}{\partial y}$                                    |  8 |  -8 |   |   |
|  $\frac{\partial^2 f}{\partial x \partial y}$                       |   24|-24   |   |   |
| $D(a,b)= \frac{\partial f}{\partial x}*\frac{\partial f}{\partial y} - (\frac{\partial^2 f}{\partial x \partial y})^2 $|  -576 |-576   |   |   |





Since -576 is a negative number and less than 0, we can say that its a saddle point. (Neither local maximum or minimum) also sine value is not equal to 0  so our 2nd derivative test is valid (inconclusive is =0).



#### 3. 

A grocery store sells two brands of a product, the “house” brand and a “name” brand. The manager estimates that if she sells the “house” brand for x dollars and the “name” brand for y dollars, she will be able to sell 81 - 21 x + 17 y units of the “house” brand and 40 + 11 x - 23 y units of the “name” brand.

Step 1. Find the revenue function R ( x, y ).

Revenue for House brand: $R(x)=x∗(81−21x+17y)$
Revenue for name brand: $R(y)=y∗(40+11x−23y)$

Total Revenue: $R(x,y)=x∗(81−21x+17y)+y∗(40+11x−23y)$

$R(x,y)=81x−21x^2+17xy+40y+11xy−23y^2$

$R(x,y)= - 21x^2 - 23y^2 + 28xy + 81x + 40y $

In [11]:
def get_rev(x,y):
 return ((- 21*x**2 )- (23*y**2) + (28*x*y) + (81*x) + (40*y))
 

Step 2. What is the revenue if she sells the "house" brand for \$2.30 and the "name" brand for \$4.10?

In [12]:
print(" \"house\" brand for $2.30 and the \"name\" brand for $4.10, wold result in revenue => ", round(get_rev(2.30,4.10),2))

 "house" brand for $2.30 and the "name" brand for $4.10, wold result in revenue =>  116.62


#### 4.
A company has a plant in Los Angeles and a plant in Denver. The firm is committed to produce a total of 96 units of a product each week. The total weekly cost is given by $C(x,y) = \frac{1}{6} x^2 + \frac{1}{6} y^2 + 7x + 25y + 700$ , where x is the number of units produced in Los Angeles and y is the number of units produced in Denver. How many units should be produced in each plant to minimize the total weekly cost?

 Number of Units from LA = x
 Number of Units from Denver = y
 x + 7 = 96 (given), we will find best value of x and y where cost can be mimunum . I.e. cost /value  of the function f(x,y) = c(c,y) at the crtical point . 

$
\textrm{Total unit per week} => x+y=96 \\
C(x,y) = \frac{1}{6}x^2 + \frac{1}{6}y^2 +7x +25y +700 \\
C(x) = \frac{1}{6}x^2 + \frac{1}{6}(96-x)^2 +7x +25(96-x) +700\\
C(x) = \frac{1}{6}x^2 + 1536-32x + \frac{1}{6}x^2 +7x +2400-25x +700 \\
C(x) = \frac{1}{3}x^2 -50x +4636 \\
\frac{dC}{dx} = \frac{2}{3}x -50 =0  ...\textrm{1st derivative}\\
x = 75 \\
\frac{d^2C}{dx^2} = \frac{2}{3}....\textrm{2nd derivative}$

Since 2/3 (2nd derivative test is >0, for x = 75 , this means that this value of x = 75 and y= 96-75= 21 is the local Minima.
This shows that 75 units in L.A. and 21 units in Denver would result in less cost.

In [15]:
def get_cost(x,y):
 return (((1/6)*𝑥**2)+(16*𝑦**2)+(7*𝑥)+(25*𝑦)+700)

get_cost(74,22),get_cost(75,21),  get_cost(76,20), 

(10424.666666666666, 9743.5, 9094.666666666668)

#### 5.
Evaluate the double integral on the given region.

$ \int \int_R (e^{8x + 3y}) dA; R: 2 \leq x \leq 4 \ { and } \ 2 \leq y \leq 4  $  
Write your answer in exact form without decimals.

$A = \int_2^4\int_2^4 e^{(8x + 3y)} dA \\
 = \int_2^4\int_2^4 e^{8x} e^{3y} dxdy \\
 = \int_2^4 e^{8x}dx * \int_2^4 e^{3y}dy \\
 = \frac{1}{8} e^{8x}\Big|_2^4 * \frac{1}{3} e^{3y}\Big|_2^4 \\
 = \frac{1}{24}e^{8x}\Big|_2^4 *  e^{3y}\Big|_2^4 \\
 = \frac{1}{24} (e^{32} - e^{16} )( e^{12} - e^{6})\\
$

ref: https://www.youtube.com/watch?v=odkaPgWPQGo
ref: https://towardsdatascience.com/simple-and-multiple-linear-regression-in-python-c928425168f9